### DISCUSSION TITLE
DPAS instruction on PVC

### DISCUSSION THREAD

**Discussion by elliottbinder** [intel/llvm#14371] [Q&A] [answered]
I'm trying to port some code that I've developed for an A770 to PVC that uses DPAS instructions. From what I gather, the difference in the interface for the DPAS instructions is that the N dimension of the matrix multiplication (referred to as systolic depth) increases from 8 to 16.

When I change this parameter to the esimd::xmx::dpas<> template, I get a static assertion error from sycl/ext/intel/esimd/xmx/dpas.hpp stating that the systolic depth must be 8. Is this a bug or is there another way I should be using DPAS instructions on PVC?

**Comment by bader** [comment]
@intel/dpcpp-esimd-reviewers, FYI.

**Reply by sarnex** [reply]
@fineg74 Do you mind taking a look at this? Thanks

**Reply by elliottbinder** [reply]
I might be mixing up systolic depth and execution width. I'll see if I can change the parameters around a bit and see if I can get things to work.

**Reply by elliottbinder** [reply, marked as answer]
Apologies, it looks like I was mixing up the dimensions. Systolic depth refers to part of the reduction K dimension, repeat count refers to M, and execution width refers to N. Execution width is implicit whereas systolic depth and repeat count are arguments to the dpas template. Reseting systolic depth back to 8 fixed this, after adjusting the esimd vector sizes appropriately for the new execution width. Thanks for the quick triage on this issue!

**Reply by sarnex** [reply]
Glad it's working!
